vyper/builtins/functions.py

```diff
@@ -229,6 +229,17 @@ def build_IR(self, expr, context):
 ADHOC_SLICE_NODE_MACROS = ["~calldata", "~selfcode", "~extcode"]
 
 
+# make sure we don't overrun the source buffer, checking for
+# overflow:
+# valid inputs satisfy: `assert start+length <= src_len && start+length > start`
+def _make_slice_bounds_check(start, length, src_len):
+    return [
```

Comment:
i'd like to see sanity check assertions that `start`, `length` and `src_len` all have `.is_complex_ir is False`